streaming/base/sampling.py

```diff
@@ -23,6 +23,18 @@ def get_sampling(samples_per_shard: NDArray[np.int64], choose: int, granularity:
     Returns:
         NDArray[np.int64]: Array of ephemeral samples chosen per shard.
     """
+    if choose < 0:
+        raise ValueError('Choose must be a non-negative integer.')
+
+    if granularity <= 0:
+        raise ValueError('Granularity must be a positive integer.')
+
+    if seed < 0:
+        raise ValueError('Seed must be a non-negative integer.')
+
+    if epoch < 0:
```

Comment:
Can this ever be zero since it is not a user provided args ?

Reply:
The first epoch is zero

For why a check for negative epoch, `epoch` is passed as an arg to this public method and it costs us near-nothing